### PR TITLE
Move suppress_tags config usage to template

### LINKED ISSUE #4681: [nsfw] Add hashtag-only hiding
Currently, the NS​FW addon only allows to hide posts containing specific words. If I set `ns​fw`, it will hide the posts with `ns​fw` and `#ns​fw`.

However, if I set `#ns​fw`, it won't hide posts with `#ns​fw`. This is due to the way we are separating the `#` from the hashtag label to make the final link.

The solution is to look in the `term` table if the provided hiding string is a hashtag.

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #4681

- Add new item.suppress_tags template variable
- Fix template formatting

Addon counter-part: https://github.com/friendica/friendica-addons/pull/560

### CODE REVIEW

File: src/Object/Post.php [hunk 1]
```diff
@@ -337,6 +338,7 @@ public function getTemplateData($conv_responses, $thread_level = 1)
 		$tmp_item = [
 			'template'        => $this->getTemplate(),
 			'type'            => implode("", array_slice(explode("/", $item['verb']), -1)),
+			'suppress_tags'   => Config::get('system', 'suppress_tags'),
```
Comment (annando) [inline src/Object/Post.php]: Shouldn't we set ```false``` as default value?

Comment (MrPetovan) [reply src/Object/Post.php]: Not necessarily, the default value would be `null` which truthiness is `false` and is enough for how we use it.

**Comment by MrPetovan** [comment]
> Why had you done the change? Isn't this more work than just not filling the templates variables?

Look at the related addon PR, the addon was looking into `$b['item']['tags']` and now `$b['item']['hashtags']` to determine if the post should be hidden or not. These variables weren’t set at all when `suppress_tags = true`.

Instead of duplicating the term fetching query in the addon, it now runs all the time and the config only affects the display at it should have done so initially in my opinion.

**Comment by annando** [comment]
Now I understand. But hadn't it be easier to create the data in "prepare_body", so that the addon has got the information and then to just nullify it before it was transmitted to the template? Instead you added the config value to the template and did the checking in every template.

**Comment by MrPetovan** [comment]
I don't think the data should be nullified, it is counter-intuitive and prone to errors, after all so far the nsfw addon was silently not checking item tags if `suppress_tags` was enabled. This prevented a post from being hidden if it had been manually tagged `#nsfw` by someone else, including the person with the nsfw filter!

There is no significant cost to checking a boolean, even in tens of template runs, so I'd rather have something intuitive.

**Pull request merged**